satpy/readers/hdfeos_base.py

```diff
@@ -148,7 +148,10 @@ def _read_mda(cls, lines, element=None):
 
     @classmethod
     def _split_line(cls, line, lines):
-        key, val = line.split("=")
+        try:
+            key, val = line.split("=")
+        except ValueError:
+            key, val = line.split("=", maxsplit=1)
```

Comment:
I'd like @mraspaud to review this as I think he wrote the original version of this code.

Reply:
While I still want @mraspaud to review this change, @simonrp84 I'm confused why codecov says your new lines aren't covered. I suppose none of your tests check this actual metadata parsing?

Reply:
I've updated the tests so that both the old and new lines are covered.

Reply:
I'm good with this, but why not just use the maxsplit argument everything? It might even make this slightly faster.

Reply:
You mean remove the try/except and only use the `maxsplit=1` version of the call? Yeah I guess that was the question, but seems obvious now. Without maxsplit=1, this will only ever work if there is a single `"="` to split on so always specifying it makes the most sense.

Reply:
OK, will change.

Reply:
Done!